frontend/src/tests/instructor-permission-test.js

```diff
@@ -53,21 +99,114 @@ export function instructorsPermissionTests(api) {
         });
     });
 
-    it.todo("can't update instructors except for self (i.e. active user)");
+    it("can't update instructors except for self (i.e. active user)", async () => {
+        // we use the default user's admin privilege here to fetch an unrelated instructor
+        let respFetchInst = await apiGET(`/instructor/instructors`);
+        expect(respFetchInst).toHaveStatus("success");
+
+        let respFetchDefaultUser = await apiGET(`/instructor/active_user`);
+        expect(respFetchDefaultUser).toHaveStatus("success");
+        const defaultUser = respFetchDefaultUser.payload;
+
+        const newDefaultUser = { ...defaultUser };
+        newDefaultUser.email = "test@test.com";
+
+        // instructor can not modify other instructors
+        await switchToInstructorOnlyUser();
+        let respInvalidRouteWithSession = await apiPOST(
+            `/instructor/instructors`,
+            newDefaultUser
+        );
+        expect(respInvalidRouteWithSession).toHaveStatus("error");
+        checkPropTypes(errorPropTypes, respInvalidRouteWithSession);
+
+        await restoreDefaultUser();
+
+        // instructors can modify themselves
+        let respModDefaultUser = await apiPOST(
+            `/instructor/instructors`,
+            newDefaultUser
+        );
+        expect(respModDefaultUser).toHaveStatus("success");
+        expect(respModDefaultUser.payload.email).toEqual("test@test.com");
+    });
 
     it("fetch sessions", async () => {
-        let resp = await apiGET("/instructor/instructors");
+        let resp = await apiGET("/instructor/sessions");
         expect(resp).toHaveStatus("success");
     });
 
-    it.todo("can't update session");
+    it("can't update session", async () => {
+        await switchToInstructorOnlyUser();
+
+        const newSession = {
+            start_date: "2020-01-01",
+            end_date: "2020-01-02",
+            name: "test",
```

Comment:
Bad name. Also, a session that lasts only one day? Make realistic sample data if you are able.